webapp/client/src/components/FormFieldTaxNumber.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import PropTypes from "prop-types";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabelForSeparatedFields from "./FieldLabelForSeparatedFields";
+import FormFieldSeparatedField from "./FormFieldSeparatedField";
+import {
+  baselineBugFix,
+  numericInputMask,
+  numericInputMode,
+} from "../lib/fieldUtils";
+
+function FormFieldTaxNumber({
+  fieldName,
+  fieldId,
+  values,
+  required,
+  autofocus,
+  details,
+  errors,
+  isSplit,
+}) {
+  const { t } = useTranslation();
+
+  const notSplitLabel = {
+    text: t("lotseFlow.taxNumber.taxNumberInput.label.labelText"),
+    exampleInput: t("lotseFlow.taxNumber.taxNumberInput.label.exampleInput"),
+  };
+
+  const splitLabel = {
+    text: t("lotseFlow.taxNumber.taxNumberInput.label.labelText"),
+  };
+
+  const fieldLabelProps = {
+    label: notSplitLabel,
+    fieldId,
+    details,
+  };
+
+  if (isSplit) {
+    fieldLabelProps.label = splitLabel;
+  }
+
+  const labelComponent = <FieldLabelForSeparatedFields {...fieldLabelProps} />;
+
+  const extraFieldProps = {
+    ...numericInputMode,
+    ...numericInputMask,
+    ...baselineBugFix,
+  };
+
+  const inputFieldLengths = isSplit ? [3, 4, 4] : [11];
+  const concatValues = isSplit ? values : [values.join()];
+
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+        labelComponent,
+      }}
+      hideLabel
+      hideErrors
+      render={() => (
+        // Separated fields ignore field class names
+        <FormFieldSeparatedField
+          {...{
+            fieldName,
+            labelComponent,
+            errors,
+            details,
+            extraFieldProps,
+            fieldId,
+            values: concatValues,
+            required,
+          }}
+          autoFocus={autofocus || Boolean(errors.length)}
+          inputFieldLengths={inputFieldLengths}
+        />
+      )}
+    />
+  );
```

Comment:
Hmm, I wouldn't have expected a `SeparatedField` when `isSplit === false`. Maybe my thinking is too literal, but wouldn't something very roughly like this be more explicit easier to understand?

```suggestion
  const splitFieldLengths = [3, 4, 4];
  return (
    <FormFieldScaffolding
      {...{
        fieldName,
        errors,
        labelComponent,
      }}
      hideLabel
      hideErrors
      render={() => (
        isSplit ? (
          <FormFieldSeparatedField
            {...{
              fieldName,
              labelComponent,
              errors,
              details,
              extraFieldProps,
              fieldId,
              values: values,
              required,
            }}
            autoFocus={autofocus || Boolean(errors.length)}
            inputFieldLengths={splitFieldLengths}
          />
        ) : (
          <FormFieldTextInput
            {...{
              fieldName,
              labelComponent,
              errors,
              details,
              extraFieldProps,
              fieldId,
              value: values.join(),
              required,
            }}
            autoFocus={autofocus || Boolean(errors.length)}
          />
        )
        )}
      />
    );
  }
```

`FormFieldTextInput` would have to be created for that, of course. :)

Reply:
Yep. However, my understanding was that we wanted to keep it a SeparatedField in the backend for both cases (split and not split). That is why I used a separated field with only one text input.

Reply:
Otherwise, we would need to flip it on the head and put it all together in the frontend.

Reply:
Sorry, I'm a bit slow these days. Why does keeping it separated in the backend entail keeping it separated in the frontend?